File: photonicdrivers/TimeTaggers/examples/python/1-Quickstart/2-controlling-the-hardware/A_get_hardware_information.py
"""In this example, we look at methods that provide information on the Time Tagger(s) in your lab."""

import TimeTagger
import pprint

available_taggers = TimeTagger.scanTimeTagger()
if available_taggers:
    print("\nTime Taggers available via TimeTagger.scanTimeTagger():")
    print(available_taggers)
else:
    print("There are no Time Taggers available. Connect one and retry.")
    exit()

tagger = TimeTagger.createTimeTagger()

print("\nYour Time Tagger model via tagger.getModel():")
print(tagger.getModel())

print("\nYour Time Tagger serial number via tagger.getSerial():")
print(tagger.getSerial())

print("\nAcquiring internal sensor data of your Time Tagger via tagger.getSensorData().")
try:
    print("\nSensor data:")
    pprint.pprint(tagger.getSensorData())
except RuntimeError:
    print("\nThe {}s do not support sensor readout.".format(tagger.getModel()))

print("\nFull Time Tagger Configuration via tagger.getConfiguration():")
pprint.pprint(tagger.getConfiguration())

TimeTagger.freeTimeTagger(tagger)

File: photonicdrivers/TimeTaggers/examples/Matlab/1-Quickstart/2-controlling-the-hardware/A_get_hardware_information.m
% In this example we look at methods that provide information on the Time Tagger(s) in your lab.

% Scan for available TimeTaggers
available_taggers = TimeTagger.scanTimeTagger();
if ~isempty(available_taggers)
    disp('Time Taggers available via TimeTagger.scanTimeTagger():');
    disp(available_taggers);
else
    disp('There are no Time Taggers available. Connect one and retry.');
    return;
end

% Create a TimeTagger instance
tagger = TimeTagger.createTimeTagger();

disp(' ');
disp('Your Time Tagger model via tagger.getModel():');
disp(tagger.getModel());

disp(' ');
disp('Your Time Tagger serial number via tagger.getSerial():');
disp(tagger.getSerial());

% Acquiring internal sensor data of your Time Tagger
disp(' ');
disp('Acquiring internal sensor data of your Time Tagger via tagger.getSensorData().');
try
    disp(' ');
    disp('Sensor data:');
    disp(tagger.getSensorData());
catch
    disp(['The ', tagger.getModel(), 's do not support sensor readout.']);
end

disp(' ');
disp('Full Time Tagger Configuration via tagger.getConfiguration():');
disp(tagger.getConfiguration());

% Free the Time Tagger instance
freeTimeTagger(tagger);
